packages/iov-core/src/integrationtests/atomicswap.spec.ts

```diff
@@ -102,175 +106,117 @@ class Actor {
-    const blockInfo = await post.blockInfo.waitFor(info => !isBlockInfoPending(info));
-    if (!isBlockInfoSucceeded(blockInfo)) {
-      throw new Error("Transaction failed");
-    }
-    await tendermintSearchIndexUpdated();
+    return this.sendTransaction(transaction);
   }
 
-  public async claimFromPreimageOnBcp(): Promise<void> {
-    // review counter
-    // TODO: search counter by swap ID
-    const searchResult = await this.bcpConnection.getSwaps({ recipient: this.bcpAddress });
-    expect(searchResult.length).toEqual(1);
-    const counterReview = searchResult[0];
-
-    if (counterReview.kind !== SwapProcessState.Open) {
-      throw new Error("Swap should be open");
-    }
-
-    if (counterReview.data.recipient !== this.bcpAddress) {
-      throw new Error("Swap counter has wrong recipient");
-    }
-
-    expect(counterReview.data.amounts.length).toEqual(1);
-    expect(counterReview.data.amounts[0].quantity).toEqual("5000000000");
-    expect(counterReview.data.amounts[0].fractionalDigits).toEqual(9);
-    expect(counterReview.data.amounts[0].tokenTicker).toEqual("MASH");
-
-    // reciew ok, alice claims MASH on BCP
-    const claim = await this.bcpConnection.withDefaultFee<SwapClaimTransaction>({
+  public async claimFromKnownPreimageOnBcp(offer: AtomicSwap): Promise<Uint8Array | undefined> {
+    const transaction = await this.bcpConnection.withDefaultFee<SwapClaimTransaction>({
       kind: "bcp/swap_claim",
       creator: this.bcpIdentity,
-      swapId: counterReview.data.id,
+      swapId: offer.data.id,
       preimage: this.preimage!,
     });
-
-    const post = await this.signer.signAndPost(claim);
-    const blockInfo = await post.blockInfo.waitFor(info => !isBlockInfoPending(info));
-    if (!isBlockInfoSucceeded(blockInfo)) {
-      throw new Error("Transaction failed");
-    }
-    await tendermintSearchIndexUpdated();
+    return this.sendTransaction(transaction);
   }
 
-  public async claimFromClaimOnBns(): Promise<void> {
-    const searchResults = await this.bcpConnection.getSwaps({ sender: this.bcpAddress });
-    expect(searchResults.length).toEqual(1);
-    const claim1Review = searchResults[0];
-
-    if (claim1Review.kind !== SwapProcessState.Claimed) {
-      throw new Error("Swap should be claimed");
-    }
-
-    // found preimage on BCP, now bob claims CASH on BNS
-    const claim2 = await this.bnsConnection.withDefaultFee<SwapClaimTransaction>({
+  public async claimFromRevealedPreimageOnBns(
+    claim: AtomicSwap,
+    unclaimedId: Uint8Array,
+  ): Promise<Uint8Array | undefined> {
+    const transaction = await this.bnsConnection.withDefaultFee<SwapClaimTransaction>({
       kind: "bcp/swap_claim",
       creator: this.bnsIdentity,
-      swapId: claim1Review.data.id,
-      preimage: claim1Review.preimage, // public data now!
+      swapId: unclaimedId as SwapIdBytes,
+      preimage: (claim as ClaimedSwap).preimage, // public data now!
```

Comment:
We should have an `isClaimedSwap` type checker instead of this cast (https://github.com/iov-one/iov-core/issues/922)